common/legacy/class.LegacyAutoLoader.php

```diff
@@ -129,6 +131,33 @@ public function autoload($pClassName)
                     return;
                 }
             }
+
+
+            $this->tryLoadFromRootFolder($pClassName);
+        }
+    }
+
+    private function tryLoadFromRootFolder($pClassName)
+    {
+        $tokens = explode("_", $pClassName);
+        if (!file_exists($this->root.'manifest.php')) {
+            return;
+        }
+        $manifest = new Manifest($this->root.'manifest.php');
+        if ($manifest->getName() !== $tokens[0]) {
+            return;
+        }
+        $path = implode('/', array_slice($tokens, 1, -1)) . '/';
+        $classFilePath = $this->root . $path . 'class.' . end($tokens) . '.php';
+        $interfaceFilePath = $this->root . $path . 'interface.' . end($tokens) . '.php';
+
+        if (file_exists($classFilePath)) {
+            require_once $classFilePath;
+            return;
+        }
+        if (file_exists($interfaceFilePath)) {
+            require_once $interfaceFilePath;
+            return;
         }
     }
```

Comment:
Considering the autoloading process is practically the same apart from the different path, I recommend the following simplifaction to avoid duplicating the logic:

```suggestion
    private function tryLoadFromRootFolder(string $extension, string $pClassName): void
    {
        if ($extension === (new Manifest($this->root . 'manifest.php'))->getName()) {
            $newClassName = preg_replace('/^[^_]+_/', '', $pClassName);
            // avoid being stuck in recursion
            ($newClassName !== $pClassName) && $this->autoload($newClassName);
        }
    }
```

To be called: 
```
$this->tryLoadFromRootFolder($token[0], $pClassName);
```

WDYT @hutnikau

Reply:
Thanks for your suggestion. I would like to avoid recursion call there and try to stick to original implementation as close as possible. This is critical part of tao and any issue is there may cause big impact.

Reply:
Code duplication leads to issues thus my recommendation. At the end of course it's your call as being an expert here 😉